### PR TITLE
Minor tweak to use useRange

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Link: -

### Problem Statement

In Gallery, Onboarding and Article page, the code always checks whether it's on the given range.

### Solution

gives useRange a range, and don't repeat the condition checking code within the main code.

### Note

Minor tweak

Onboarding code doesn't change much, and this is ready for the upcoming onboarding change.

### CODE REVIEW

File: src/hooks/useArticlePagination.js [hunk 1]
```diff
@@ -69,18 +68,12 @@ export const useArticlePagination = (
...
   }
 
   const showPrevSection = () => {
-    const prevSection = currentSection - 1
-    if (prevSection >= 0) {
-      setCurrentSection(prevSection)
-      setIsLastPage(true)
-    }
+    setCurrentSection(currentSection - 1)
+    setIsLastPage(true)
```
Comment (hueitan) [inline src/hooks/useArticlePagination.js]: oh wait, this need to be guard.

File: src/hooks/useRange.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  const setRangeValue = newValue => {
+    if (newValue < min) {
+      setValue(min)
+    } else if (newValue > max) {
+      setValue(max)
+    } else {
+      setValue(newValue)
+    }
+  }
+  return [value, setRangeValue]
```
Comment (stephanebisson) [inline src/hooks/useRange.js]: Would it make sense for `useRange` to expose `next()` and `previous()` functions?

Comment (hueitan) [reply src/hooks/useRange.js]: Good comment! It's totally make sense and that's the idea why another hook useRange here.

**Comment by hueitan** [comment]
ready to be reviewed now.

**Pull request merged**